added reference for crawler stats (scheduler, downloader, logs)

musharafmaqbool opened the pull request:
This PR adds a new documentation page that lists the statistics exposed through crawler.stats.

Currently, many of these stats are used internally and appear in logs, but are not described anywhere in the docs. This makes it difficult for users to interpret crawl statistics.

The new page documents:

Log counters (DEBUG, INFO, WARNING, ERROR)
Scheduler stats (enqueued/dequeued, memory subsets)
Downloader stats (request/response counts, status codes, bytes)
Other useful stats (request_depth_max, start_time, item counts)
This change addresses issue https://github.com/scrapy/scrapy/issues/6351 and provides users with a clear reference for interpreting Scrapy’s runtime stats.

Resolves https://github.com/scrapy/scrapy/issues/6351.

wRAR commented on docs/topics/stats-reference.rst:
This page is not referenced anywhere.

wRAR commented on docs/topics/stats-reference.rst:
```diff
@@ -0,0 +1,60 @@
+Scheduler
+=========
+
+- *scheduler/enqueued*: number of requests placed into the scheduler.
+- *scheduler/enqueued/memory*: requests kept in memory by the scheduler.
+- *scheduler/dequeued*: number of requests pulled from the scheduler.
+- *scheduler/dequeued/memory*: requests pulled from memory by the scheduler.
```
I think these descriptions are unclear and/or incorrect.

musharafmaqbool commented:
@wRAR I've updated the documentation to address your feedback:

1. Made it much more comprehensive - now covers stats from various middlewares, extensions, and error scenarios, not just from sample runs
2. Fixed the scheduler stat descriptions to be clearer and more accurate
3. Added practical usage examples and better organization
4. Included production monitoring guidance

Please let me know if this addresses your concerns or if you'd like any other changes.

wRAR commented on docs/topics/stats-reference.rst:
```diff
@@ -0,0 +1,182 @@
+AutoThrottle Extension
+----------------------
+
+* **autothrottle/response_interval_avg**: average response time used for throttling calculations.
```
Can you please show where did you find this?

musharafmaqbool commented:
@wRAR sure:

**Primary Sources:**
1. **Scrapy source code analysis**: I reviewed the core Scrapy modules where stats are generated:
   - `scrapy/core/scheduler.py` - for scheduler/* stats
   - `scrapy/core/downloader/` - for downloader/* stats  
   - `scrapy/spiders/__init__.py` - for spider execution stats
   - `scrapy/dupefilters.py` - for dupefilter/* stats

2. **Extension and middleware source code**:
   - `scrapy/extensions/memusage.py` - for memusage/* stats
   - `scrapy/extensions/autothrottle.py` - for autothrottle/* stats
   - `scrapy/downloadermiddlewares/retry.py` - for retry/* stats
   - `scrapy/downloadermiddlewares/redirect.py` - for redirect/* stats
   - `scrapy/downloadermiddlewares/httpcache.py` - for httpcache/* stats

3. **Testing with actual crawls**: I ran test spiders with various configurations to observe which stats appear under different conditions.

4. **Existing Scrapy documentation**: Referenced the current stats documentation and built upon it.

**Specific examples I can point to:**
- Lines in `scrapy/core/scheduler.py` where `crawler.stats.inc_value('scheduler/enqueued')` is called
- The `memusage` extension code that creates the memory-related stats
- The retry middleware that increments `retry/count` stats
